### ISSUE TITLE
iOS Build Fails Due to Sentry Integration

### ISSUE THREAD

**Issue by victorfob** [getsentry/sentry-react-native#4100]
**OS:**
- [ ]  Windows
- [X]  MacOS
- [ ]  Linux

**Platform:**
- [X]  iOS
- [ ]  Android

**SDK:**
- [X]  `@sentry/react-native` (>= 1.0.0)
- [ ]  `react-native-sentry` (<= 0.43.2)

**SDK version:** 5.24.2

**`react-native` version:** 0.74.5
**`Xcode` version:** 16.0
**`MacOS` version:** 15.0
**`Expo`version:** 51.0.31

**Are you using Expo?**
- [X]  Yes
- [ ]  No

**Are you using sentry.io or on-premise?**
- [X]  sentry.io (SaaS)
- [ ]  on-premise

**Configuration:**

Not applicable, the error occurs during the build phase.

---
**I have the following issue:**

When attempting to build my Expo managed project for iOS using the command `npx expo run:ios`, the build process fails. When I try to build the iOS project directly in Xcode, I encounter the following error:

```bash
/Users/MY_USER/Development/Projects/project-name/ios/Pods/Sentry/Sources/SentryCrash/Recording/Tools/SentryCrashMachineContext.c:43:9: 
Missing '#include <sys/_types/_ucontext64.h>'; 'ucontext64_t' must be declared before it is used
```
![Image](https://github.com/user-attachments/assets/82ce8aba-2dd9-4654-bb47-32af7aee55e9)

After manually including the missing line, the build completes successfully using the same command.

Additionally, if I remove Sentry from the project dependencies, the build proceeds without any issues.

**Steps to reproduce:**
- Set up an Expo managed project with Sentry.
- Attempt to build the iOS project using the command:
```bash
npx expo run:ios
```

**Actual result:**

```bash
** BUILD FAILED **

The following build commands failed:
        CompileC /Users/MY_USER/Library/Developer/Xcode/DerivedData/SLUG-CODE/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/Sentry.build/Objects-normal/arm64/SentryCrashMachineContext.o /Users/MY_USER/Development/Projects/project-name/ios/Pods/Sentry/Sources/SentryCrash/Recording/Tools/SentryCrashMachineContext.c normal arm64 c com.apple.compilers.llvm.clang.1_0.compiler (in target 'Sentry' from project 'Pods')
        Building workspace SLUG with scheme SCHEME and configuration Debug
(2 failures)
```

**Expected result:**

The iOS build should complete successfully without errors.

**Comment by kahest** [comment]
Hey @victorfob this is a duplicate of https://github.com/getsentry/sentry-react-native/issues/4097 and a fix is available by upgrading the SDK to 5.31.0 or newer.

**Comment by victorfob** [comment]
> Hey [@victorfob](https://github.com/victorfob) this is a duplicate of [#4097](https://github.com/getsentry/sentry-react-native/issues/4097) and a fix is available by upgrading the SDK to 5.31.0 or newer.

Would it be possible to release a backwards-compatible update, such as version 5.24.3? The reason I ask is that I use Expo, and upgrading to newer SDK versions could potentially introduce other issues since they might not be fully compatible with Expo yet.

Thanks!

**Comment by kahest** [comment]
@victorfob we released 5.24.3 with a bump to sentry-cocoa: https://github.com/getsentry/sentry-react-native/releases/tag/5.24.3
